Torch.Server/ViewModels/ConfigDedicatedViewModel.cs

```diff
@@ -73,12 +75,57 @@ public WorldViewModel SelectedWorld
             }
         }
 
+        public async Task UpdateAllModInfosAsync(Action<string> messageHandler = null)
+        {
+            if (Mods.Count() == 0)
+                return;
+
+            var ids = Mods.Select(m => m.PublishedFileId);
+            var workshopService = WebAPI.Instance;
+            Dictionary<ulong, PublishedItemDetails> modInfos = null;
+
+            try
+            {
+                modInfos = (await workshopService.GetPublishedFileDetails(ids.ToArray()));
+            }
+            catch (Exception e)
+            {
+                Log.Error(e.Message);
+                return;
+            }
+
+            Log.Info($"Mods Info successfully retrieved!");
+
+            foreach (var mod in Mods)
```

Comment:
Not threadsafe. Mods is a straight ObservableCollection, and may be accessed by multiple threads concurrently. Mods collection must be changed to a threadsafe collection type.

Reply:
Increases the number of allocations, but should be thread safe :P

Reply:
.ToArray will drastically reduce the number of MT exceptions, but not completely eliminate them. DedicatedConfig.Mods should be changed to type MtObservableCollection.

Reply:
Implemented in commit [917e6fd](https://github.com/TorchAPI/Torch/pull/263/commits/917e6fddd375f81115fd6c4478d8edaa00e73827)